bugbug/tools/release_notes.py

```diff
@@ -0,0 +1,250 @@
+import logging
+import re
+from itertools import batched
+from typing import Generator, Optional
+
+import requests
+from langchain.chains import LLMChain
+from langchain.prompts import PromptTemplate
+
+from bugbug import bugzilla, db
+
+KEYWORDS_TO_REMOVE = [
+    "Backed out",
+    "a=testonly",
+    "DONTBUILD",
+    "add tests",
+    "disable test",
+    "back out",
+    "backout",
+    "add test",
+    "added test",
+    "ignore-this-changeset",
+    "CLOSED TREE",
+    "nightly",
+]
+
+PRODUCT_OR_COMPONENT_TO_IGNORE = [
+    "Firefox Build System::Task Configuration",
+    "Developer Infrastructure::",
+]
+
+
+def get_previous_version(current_version: str) -> str:
+    match = re.search(r"(\d+)", current_version)
+    if not match:
+        raise ValueError("No number found in the version string")
+
+    number = match.group(0)
+    decremented_number = str(int(number) - 1)
+    return (
+        current_version[: match.start()]
+        + decremented_number
+        + current_version[match.end() :]
+    )
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class ReleaseNotesCommitsSelector:
+    def __init__(self, chunk_size: int, llm: LLMChain):
+        self.chunk_size = chunk_size
+        self.bug_id_to_component = {}
+        db.download(bugzilla.BUGS_DB)
```

Comment:
It is too much to download the whole bugs database to check a small set of them. We could instead deal with the ones that we need by their IDS. 

See: https://github.com/mozilla/bugbug/blob/bba1d097f968ef18ec5687f7e3064c792660c702/bugbug/bugzilla.py#L219

Reply:
Even better, you could use `libmozdata.bugzilla` to request just the bugs you need, including only the necessary fields.